src/tree_cmd.cpp

```diff
@@ -102,6 +102,7 @@ static void PlantTreesOnTile(TileIndex tile, TreeType treetype, uint count, Tree
 	switch (GetTileType(tile)) {
 		case MP_WATER:
 			ground = TREE_GROUND_SHORE;
+			ClearNeighbourNonFloodingStates(tile);
```

Comment:
From my understanding, this shouldn't be here when placing trees, but instead it should be when clearing the tree tile, in the same vein as when clearing a coast tile.

Reply:
This re-enables flooding on water tiles next to the tree tile. Related to why `continue_flooding` is set before the `TREE_GROUND_SHORE` test.